tests/chainer_tests/functions_tests/array_tests/test_hstack.py

```diff
@@ -61,4 +66,34 @@ def test_backward_gpu(self):
             [cuda.to_gpu(x) for x in self.xs], cuda.to_gpu(self.g))
 
 
+@testing.parameterize(
+    {'a_shape': (3, 4), 'b_shape': (4, 4), 'valid': False},
+    {'a_shape': (3, 4), 'b_shape': (3, 3), 'valid': True},
+    {'a_shape': (3,), 'b_shape': (4,), 'valid': True},
+    {'a_shape': (3), 'b_shape': (3, 3), 'valid': False},
+)
```

Comment:
Could you add the test case where at least one of `a_shape` and `b_shape` are longer than 2?